File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        arr = []
        vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        
        for i in range(len(s)):
            if(s[i] in vowels):
                arr.append(s[i])
        s = list(s)
        count = 0
        for x in reversed(range(len(s))):
            if(s[x] in vowels):
                s[x] = arr[count]
                count += 1

        return ''.join(s)

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    let arr = []
    let vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
    for(let i = 0; i < s.length; i++){
        if(vowels.includes(s[i])){
            arr.push(s[i])
        }
    }
    s = s.split('')
    let count = 0
    for(let g = s.length; g >= 0; g--){
        if(vowels.includes(s[g])){
            s[g] = arr[count]
            count++
        }
    }
    
    return s.join('')
};